### PR TITLE
Feat handle tx errors

### PR DESCRIPTION
**Pull request by bendanzhentan**
Resolve https://github.com/bnb-chain/opbnb-bridge-bot/pull/1#discussion_r1399246313

### CODE REVIEW

**Review by owen-reorg — APPROVED** [review]

File: cmd/bot/run.go [hunk 1]
```diff
@@ -108,15 +108,21 @@ func ProcessUnprovenBotDelegatedWithdrawals(ctx context.Context, log log.Logger,
...
 					log.Error("failed to update proven l2_contract_events", "error", result.Error)
-					return
 				}
 			} else if strings.Contains(err.Error(), "L2OutputOracle: cannot get output for a block that has not been proposed") {
 				// Since the unproven withdrawals are sorted by the on-chain order, we can break here because we know
 				// that the subsequent of the withdrawals are not ready to be proven yet.
 				return
+			} else if strings.Contains(err.Error(), "execution reverted") {
+				// Proven transaction reverted, mark it with the failure reason
+				result := db.Model(&unproven).Update("failure_reason", err.Error())
```
Comment (welkin22) [inline cmd/bot/run.go:118]: Current failure_reason is set to varchar(256), but it cannot be determined that err.Error() will always be within 256 characters. It is recommended to truncate it.

Comment (bendanzhentan) [reply cmd/bot/run.go:118]: Make sense!
Updated based on your suggestion at [c463328](https://github.com/bnb-chain/opbnb-bridge-bot/pull/8/commits/c463328a76a5fc03836f8c42f7b013dd1389de42)

Comment (welkin22) [reply cmd/bot/run.go:118]: Line 161 has the same problem.

Comment (bendanzhentan) [reply cmd/bot/run.go:118]: Oh my fault.
I updated and force-push at [17ef561](https://github.com/bnb-chain/opbnb-bridge-bot/pull/8/commits/17ef561113af38a5e815d99c11163281992784a8).

File: cmd/bot/run.go [hunk 2]
```diff
@@ -318,3 +330,14 @@ func queryL2ScannedBlock(db *gorm.DB, cfg *core.Config) (*core.L2ScannedBlock, e
 	}
 	return &l2ScannedBlock, nil
 }
+
+// NOTE: types.L2ContractEvent.FailureReason is defined as `type:varchar(256)` in database.
```
Comment (owen-reorg) [inline cmd/bot/run.go]: why don't just use `text` as the type?

Comment (bendanzhentan) [reply cmd/bot/run.go]: Good suggestion.

Both truncating into `varchar(255)` and defining in `text` are good to me. However, `text` seems better. I will change to `text`.

Comment (bendanzhentan) [reply cmd/bot/run.go]: Updated at [2255cdc](https://github.com/bnb-chain/opbnb-bridge-bot/pull/8/commits/2255cdc946f77d2283f44bddc26fa429d2d32400)

**Pull request merged**